astropy/time/core.py

```diff
@@ -403,17 +403,6 @@ def __init__(self, val, val2=None, format=None, scale=None,
                  precision=None, in_subfmt=None, out_subfmt=None,
                  location=None, copy=False):
 
-        if location is not None:
```

Comment:
This was being run even for initializing from an existing `Time` object, in which case `self` already has a `location`.  But actually looking at this now maybe I need to just put something in the `else` block at the end to check if `location` already exists and not override.  That allows overriding the existing location, e.g. `Time(tm_object, location=...)`.

Reply:
Done.